esm_parser/esm_parser.py

```diff
@@ -579,6 +579,25 @@ def deep_update(chapter, entries, config, blackdict={}):
      else:
         if chapter not in blackdict:
             dict_merge(config, {chapter: entries})
+     
+        # deniz: bugfix: choose_ blocks with "*" keys don't get updated. Eg. 
+        # mail1 and mail2 don't get updated since they are initialized as empty
+        # strings and are already inside. Current bug fix only correct scalar
+        # types such as strings.  
+        else:
+            empty_values = ["", None]  # some possible empty values to override
+
+            # strip all whitespace. Eg. "  " -> "", if it is only space
+            if (isinstance(blackdict[chapter], str) and 
+                blackdict[chapter].isspace()):
+                blackdict[chapter] = re.sub(r"\s+", "", blackdict[chapter])
```

Comment:
Why do you use that syntax instead of just doing `blackdict[chapter] = ""` (I feel like I am about to learn something new :) )

Reply:
First the second part:
I added regex as a second commit, because I have a feeling in my  gut that at some point a variable will be initialized as 
`foo = " "` and `blackdict[chapter] = ""`. Here, I check if it is only whitespace (eg. combination of ordinary spaces and tabs).

First part:
This is interesting to know. Basically, in the code the `else` branch was missing, not even a debug print was written. So, I thought that it is a potential bug. 
Could you please suggest me some coupled setup? I am currently working on AWICM-1 is that Ok or do I need FESOM-Recom, ...?

Reply:
I think the `else` was missing because the `blackdict` is supposed to protect whatever variables it contains from being changed in a `deep_merge`. As far as I understand `blackdict` is a very important feature to control the overwriting of variables, so I'm wondering, even if this works, isn't it too specific? I'm a bit lost here, I also don't want to stop this from moving forward, but I have the impression that there is a more general problem behind this particular case. @dbarbi and @pgierz , do you have any extra thoughts?